src/main/java/org/apache/commons/text/similarity/LevenshteinDistance.java

```diff
@@ -285,6 +286,11 @@ private static int limitedCompare(CharSequence left, CharSequence right, final i
                     // left and up
                     d[i] = 1 + Math.min(Math.min(d[i - 1], p[i]), p[i - 1]);
                 }
+                lowerBound = Math.min(lowerBound, d[i]);
+            }
+            // if the lower bound is greater than the threshold, then exit early
+            if (lowerBound > threshold) {
+                return -1;
```

Comment:
@vesterstroem it looks to me like this nullifies the `if (p[n] <= threshold) {` further down.

I was looking at the [coverage report](https://coveralls.io/builds/33884935/source?filename=src/main/java/org/apache/commons/text/similarity/LevenshteinDistance.java), and it seems to be that the coverage decreased because that `return -1` never happens.

If so, I think we can remove it and simply `return p[n];`. WDYT?

Reply:
The improvement introduces the need for an extra test, which I forgot to add.

The assertion:

        assertThat(new LevenshteinDistance(1).apply("abc", "acb")).isEqualTo(-1);

should be inserted into e.g. testGetLevenshteinDistance_StringStringInt of LevenshteinDistanceTest.

That will give coverage to the uncovered line - and it will fail, if we just always return p[n].

Should I make a new pull request or update the existing one?

Reply:
Better update this one. We can squash/edit commits later if necessary :+1: After you push a new commit I think the CI jobs will run again, and coveralls will confirm the code is being tested now.

Thanks!